src/components/core/PivotTable.tsx

```diff
@@ -428,42 +432,90 @@ export class PivotTableInner extends BaseVisualization<IPivotTableInnerProps, IP
     private getColumnIds = (columns: Column[]): string[] =>
         columns.map((column: Column) => column.getColId());
 
-    private getResizedColumns = (columns: Column[]): IResizedColumns => {
-        const initialValue: IResizedColumns = {};
+    private getResizedColumns = (columns: Column[], source: ColumnEventSourceType): IResizedColumns => {
         return columns.reduce((acc, col) => {
+            const columnId = this.getColumnIdentifier(col.getDefinition() as IGridHeader);
+            const resizedColumn = acc[columnId];
+            if (resizedColumn) {
+                return acc;
+            }
             return {
                 ...acc,
-                [this.getColumnIdentifier(col.getDefinition() as IGridHeader)]: {
+                [columnId]: {
                     width: col.getActualWidth(),
+                    source,
                 },
             };
-        }, initialValue);
+        }, this.resizedColumns);
     };
 
-    private autoresizeVisibleColumns = (columnApi: ColumnApi, previouslyResizedColumnIds: string[]) => {
+    private addToResizeColumn = (column: Column, source: ColumnEventSourceType): void => {
+        this.resizedColumns[this.getColumnIdentifier(column.getDefinition() as IGridHeader)] = {
+            width: column.getActualWidth(),
+            source,
+        };
+    };
+
+    private autoresizeVisibleColumns = async (
+        columnApi: ColumnApi,
+        previouslyResizedColumnIds: string[],
+        firstCall: boolean = true,
+    ): Promise<void> => {
+        const { execution } = this.state;
+
+        const tableIsNotScrolled = () => {
+            const horizontalPixelRange = this.gridApi.getHorizontalPixelRange();
+            const verticalPixelRange = this.gridApi.getVerticalPixelRange();
+            return horizontalPixelRange.left === 0 && verticalPixelRange.top === 0;
+        };
+
+        if (!(execution && tableIsNotScrolled())) {
```

Comment:
What about some refactoring here with this condition. You know, when I saw it, it was not clear for me in the first place. Maybe we could have here a call similar to "isColumnAutoresizeEnabled" where will be all these checks.

Reply:
Done